movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+    interindividual distances). The distances are computed using the
+    specified ``metric``.
+
+    Parameters
+    ----------
+    data : xarray.DataArray
+        The input data containing
+        ``('time', 'individuals', 'keypoints', 'space')`` as dimensions.
+    dim : str
+        The dimension to compute the distances for. Must be either
+        ``'individuals'`` or ``'keypoints'``.
+    pairs : dict[str, str | list[str]], optional
+        A dictionary containing the mapping between pairs of ``individuals``
+        or ``keypoints``. The key is the keypoint or individual and the
+        value is a list of keypoints or individuals or a string
+        representing a keypoint or individual to compute the distance with.
+        If not provided, defaults to ``None`` and all possible combinations
+        of pairs are computed.
+    metric : str, optional
+        The distance metric to use. Must be one of the options supported
+        by :func:`scipy.spatial.distance.cdist`, i.e.
+        ``'braycurtis'``, ``'canberra'``, ``'chebyshev'``, ``'cityblock'``,
+        ``'correlation'``, ``'cosine'``, ``'dice'``, ``'euclidean'``,
+        ``'hamming'``, ``'jaccard'``, ``'jensenshannon'``, ``'kulczynski1'``,
+        ``'mahalanobis'``, ``'matching'``, ``'minkowski'``,
+        ``'rogerstanimoto'``, ``'russellrao'``, ``'seuclidean'``,
+        ``'sokalmichener'``, ``'sokalsneath'``, ``'sqeuclidean'``, ``'yule'``.
+        Defaults to ``'euclidean'``.
+    **kwargs : dict
+        Additional keyword arguments to pass to
+        :func:`scipy.spatial.distance.cdist`.
+
+    Returns
+    -------
+    xarray.DataArray | dict[str, xarray.DataArray]
+        An :class:`xarray.DataArray` containing the computed distances between
+        a single pair of ``individuals`` or ``keypoints``. If multiple
+        ``pairs`` are provided, this will be a dictionary containing the
+        computed distances for each pair, with the key being the pair of
+        keypoints or individuals and the value being the
+        :class:`xarray.DataArray` containing the computed distances.
+
+    See Also
+    --------
+    cdist : ``movement``'s wrapper around :func:`scipy.spatial.distance.cdist`.
+    :func:`scipy.spatial.distance.cdist` : The underlying function used.
+
+    """
+    if dim not in ["individuals", "keypoints"]:
+        raise log_error(
+            ValueError,
+            "'dim' must be either 'individuals' or 'keypoints', "
+            f"but got {dim}.",
+        )
+    pairwise_distances = {}
+
+    # Compute all possible pair combinations if not provided
+    if pairs is None:
+        paired_elements = list(
+            itertools.combinations(getattr(data, dim).values, 2)
+        )
+    else:
+        paired_elements = [
+            (elem1, elem2)
+            for elem1, elem2_list in pairs.items()
+            for elem2 in (
+                [elem2_list] if isinstance(elem2_list, str) else elem2_list
+            )
+        ]
```

Comment:
I had a hard time parsing this list comprehension, so I can't quite figure out the answer to the following question:

What happens if one passed duplicate pairs - e.g.`pairs={"id1": "id2", "id2": "id1"}`? We wouldn't want to compuute that twice.

If this is not currently handled, we could set-ify the `paired_elements` list before iterating over it.

Reply:
This is actually allowed as I expect users to get what they ask for (no surprises) 🤔 wdyt?

Reply:
sure, fine with that as well.